x-pack/legacy/plugins/file_upload/public/components/index_settings.js

```diff
@@ -8,17 +8,38 @@ import React, { Fragment, Component } from 'react';
 import { i18n } from '@kbn/i18n';
 import { EuiFormRow, EuiFieldText, EuiSelect, EuiCallOut } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import { getExistingIndices, getExistingIndexPatterns }
-  from '../util/indexing_service';
+import {
+  getExistingIndexNames,
+  getExistingIndexPatternNames,
+  checkIndexPatternValid,
+} from '../util/indexing_service';
 
 export class IndexSettings extends Component {
 
   state = {
     indexNameError: '',
     indexDisabled: true,
-    indexPatterns: null,
-    indexNames: null,
     indexName: '',
+    indexNameList: [],
+    indexPatternList: [],
+  };
+
+  async componentDidMount() {
+    this._isMounted = true;
+    this.loadExistingIndexData();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  loadExistingIndexData = async () => {
+    if (this._isMounted) {
```

Comment:
The check for this._isMounted needs to happen after the await. You are protected against setting state on an unmounted component.

The reason is that while the async action is happening, the user could do something else that unmounts the component before the results are returned

```
  const indexNames = await getExistingIndexNames();
  const indexPatterns = await getExistingIndexPatternNames();
  if (this._isMounted) {
   // set state
  }
```